packages/yew-router/src/history.rs

```diff
@@ -0,0 +1,237 @@
+use std::cell::RefCell;
+use std::rc::Rc;
+
+use crate::utils::base_url;
+use gloo::events::EventListener;
+use wasm_bindgen::JsValue;
+use web_sys::EventTarget;
+use yew::utils::window;
+use yew::Callback;
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct Route {
+    pub(crate) path: String,
+    pub(crate) query: String,
+    pub(crate) hash: String,
+}
+
+impl Default for Route {
+    fn default() -> Self {
+        Self {
+            path: Default::default(),
+            query: Default::default(),
+            hash: Default::default(),
+        }
+    }
+}
```

Comment:
Why does this impl exist when new takes a path parameter? Is `""` a valid path? I think we should remove the `Default` impl.

Reply:
It's used to construct an `empy` route. This is used as a starting point when building up a route from nothing, and also is used as a placeholder when the munchers temporarily take ownership of the route to modify it.

However, you're right that `Route` itself does not need to implement `Default`, so I'll replace it with a private method.

Reply:
Right. There shouldn't be a way to construct a `Route` outside of our crate.

Reply:
Now that the history API is private, you can't even name the `Route` type.